src/lib/propagation/tests/DAG-test.js

```diff
@@ -0,0 +1,140 @@
+import {expect} from 'chai'
+
+import {SimulationDAG} from '../DAG'
+import * as constants from '../constants'
+
+import * as _collections from 'gEngine/collections'
+
+// ERRORS:
+const {
+  NODE_TYPES,
+  ERROR_TYPES: {GRAPH_ERROR},
+  ERROR_SUBTYPES: {GRAPH_SUBTYPES: {MISSING_INPUT_ERROR, IN_INFINITE_LOOP, INVALID_ANCESTOR_ERROR}},
+} = constants
+
+const toExprId = id => `\$\{${toNodeId(id)}\}`
+const toNodeId = idNum => `node:${idNum}`
+const toExpr = inputNums => `=${inputNums.map(toExprId).join(' + ')}`
+const makeNode = (idNum, inputNums=[], samples=[], errors=[], type=NODE_TYPES.FUNCTION, expression=toExpr(inputNums)) => (
+  {id: toNodeId(idNum), errors, samples, type, expression}
+)
+
+const expectNodesToBe = (DAG, validIds, errorIds) => {
+  expect(DAG.graphErrorNodes.map(n => n.id)).to.deep.equal(errorIds.map(toNodeId))
+  expect(DAG.nodes.map(n => n.id)).to.deep.equal(validIds.map(toNodeId))
+}
+
+describe('construction', () => {
+  const simpleValidNodeList = [makeNode(1), makeNode(2, [1,3]), makeNode(3, [1])]
+  it ('Yields nodes ordered by height', () => {expectNodesToBe(new SimulationDAG(simpleValidNodeList), [1,3,2], [])})
+  it ('Correclty assignes nodes their relations', () => {
+    const DAG = new SimulationDAG(simpleValidNodeList)
+
+    expect(DAG.nodes.map(n => _.pick(n, ['parentIndices']))).to.deep.have.members([
```

Comment:
are the inputs parents or children?  I would have expected the inputs to be parents, but this seems like the outputs are the parents.